src/controllers/expensesControllers.js

```diff
@@ -0,0 +1,120 @@
+'use strict';
+
+const expensesService = require('../services/expensesService');
+const usersService = require('../services/usersService');
+
+const getAllExpenses = (req, res) => {
+  const { userId, from, to, categories } = req.query;
+
+  let expenses = expensesService.getAllExpenses();
+
+  if (userId) {
+    expenses = expenses.filter((e) => e.userId === +userId);
+  }
+
+  if (categories) {
+    expenses = expenses.filter((expense) =>
+      categories.includes(expense.category)
+    );
+  }
+
+  if (from) {
+    expenses = expenses.filter((e) => e.spentAt >= from);
+  }
+
+  if (to) {
+    expenses = expenses.filter((e) => e.spentAt <= to);
+  }
+
+  res.send(expenses);
+};
+
+const createExpense = (req, res) => {
+  const { userId, spentAt, title, amount, category, note } = req.body;
+
+  const idForUser = +userId;
+
+  const userById = usersService.getUserById(idForUser);
+
+  if (!userById) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  const newExpense = {
+    id: +new Date(),
+    userId,
+    spentAt,
+    title,
+    amount,
+    category,
+    note,
+  };
+
+  expensesService.addExpense(newExpense);
+
+  res.statusCode = 201;
```

Comment:
You can create a new variable for all statusCode and use it, i think

Reply:
agreed, better create constant for statuses